### ISSUE TITLE
Anthropic api not supporting

### ISSUE THREAD

**Issue by dannylank** [paul-gauthier/aider#696]
### Issue

` aider --no-auto-commits --opus --anthropic-api-key
usage: aider [-h] [--openai-api-key OPENAI_API_KEY] [--model MODEL] [-3] [--openai-api-base OPENAI_API_BASE]
             [--openai-api-type OPENAI_API_TYPE] [--openai-api-version OPENAI_API_VERSION]
             [--openai-api-deployment-id OPENAI_API_DEPLOYMENT_ID] [--openai-api-engine OPENAI_API_ENGINE] [--edit-format EDIT_FORMAT]
             [--map-tokens MAP_TOKENS] [--input-history-file INPUT_HISTORY_FILE] [--chat-history-file CHAT_HISTORY_FILE] [--dark-mode]
             [--light-mode] [--pretty] [--no-pretty] [--no-stream] [--user-input-color USER_INPUT_COLOR]
             [--tool-output-color TOOL_OUTPUT_COLOR] [--tool-error-color TOOL_ERROR_COLOR] [--assistant-output-color ASSISTANT_OUTPUT_COLOR]
             [--code-theme CODE_THEME] [--show-diffs] [--no-git] [--auto-commits] [--no-auto-commits] [--dirty-commits] [--no-dirty-commits]
             [--dry-run] [--version] [--apply FILE] [--yes] [-v] [--show-repo-map] [--message COMMAND] [-c CONFIG_FILE]
             [FILE [FILE ...]]
aider: error: unrecognized arguments: --opus --anthropic-api-key`

### Version and model info

Name: aider-chat
Version: 0.12.0
Summary: aider is GPT powered coding in your terminal
Home-page: https://github.com/paul-gauthier/aider
Author: None
Author-email: None
License: None
Location: /home/danriv/proyectos/okrai/env/lib/python3.8/site-packages
Requires: backoff, configargparse, scipy, tqdm, diskcache, requests, GitPython, tiktoken, jsonschema, aiosignal, idna, mdurl, networkx, wcwidth, numpy, sounddevice, smmap, multidict, certifi, pytest, gitdb, urllib3, Pygments, frozenlist, PyYAML, async-timeout, markdown-it-py, soundfile, yarl, openai, aiohttp, rich, prompt-toolkit, charset-normalizer, attrs

**Comment by paul-gauthier** [comment]
Thanks for trying aider and filing this issue.

Looks like you are using a very old version of aider (0.12.0). You should upgrade to the current v0.38.0.

```
pip install --upgrade aider-chat
```

**Comment by dannylank** [comment]
I'm using Ubuntu in WSL. I run pip install aider-chat and it installs version 0.12.0 
`(env) danriv@pb0b0959853:~/proyectos/typescript$ pip install --upgrade aider-chat
Requirement already satisfied: aider-chat in ./env/lib/python3.8/site-packages (0.12.0)
Requirement already satisfied: aiohttp==3.8.4 in ./env/lib/python3.8/site-packages (from aider-chat) (3.8.4)
Requirement already satisfied: aiosignal==1.3.1 in ./env/lib/python3.8/site-packages (from aider-chat) (1.3.1)
Requirement already satisfied: async-timeout==4.0.2 in ./env/lib/python3.8/site-packages (from aider-chat) (4.0.2)
Requirement already satisfied: attrs==23.1.0 in ./env/lib/python3.8/site-packages (from aider-chat) (23.1.0)
Requirement already satisfied: certifi==2023.5.7 in ./env/lib/python3.8/site-packages (from aider-chat) (2023.5.7)
Requirement already satisfied: charset-normalizer==3.1.0 in ./env/lib/python3.8/site-packages (from aider-chat) (3.1.0)
Requirement already satisfied: frozenlist==1.3.3 in ./env/lib/python3.8/site-packages (from aider-chat) (1.3.3)
Requirement already satisfied: gitdb==4.0.10 in ./env/lib/python3.8/site-packages (from aider-chat) (4.0.10)
Requirement already satisfied: GitPython==3.1.31 in ./env/lib/python3.8/site-packages (from aider-chat) (3.1.31)
Requirement already satisfied: idna==3.4 in ./env/lib/python3.8/site-packages (from aider-chat) (3.4)
Requirement already satisfied: markdown-it-py==2.2.0 in ./env/lib/python3.8/site-packages (from aider-chat) (2.2.0)
Requirement already satisfied: mdurl==0.1.2 in ./env/lib/python3.8/site-packages (from aider-chat) (0.1.2)
Requirement already satisfied: multidict==6.0.4 in ./env/lib/python3.8/site-packages (from aider-chat) (6.0.4)
Requirement already satisfied: openai==0.27.6 in ./env/lib/python3.8/site-packages (from aider-chat) (0.27.6)
Requirement already satisfied: prompt-toolkit==3.0.38 in ./env/lib/python3.8/site-packages (from aider-chat) (3.0.38)
Requirement already satisfied: Pygments==2.15.1 in ./env/lib/python3.8/site-packages (from aider-chat) (2.15.1)
Requirement already satisfied: requests==2.30.0 in ./env/lib/python3.8/site-packages (from aider-chat) (2.30.0)
Requirement already satisfied: rich==13.3.5 in ./env/lib/python3.8/site-packages (from aider-chat) (13.3.5)
Requirement already satisfied: smmap==5.0.0 in ./env/lib/python3.8/site-packages (from aider-chat) (5.0.0)
Requirement already satisfied: tqdm==4.65.0 in ./env/lib/python3.8/site-packages (from aider-chat) (4.65.0)
Requirement already satisfied: urllib3==2.0.2 in ./env/lib/python3.8/site-packages (from aider-chat) (2.0.2)
Requirement already satisfied: wcwidth==0.2.6 in ./env/lib/python3.8/site-packages (from aider-chat) (0.2.6)
Requirement already satisfied: yarl==1.9.2 in ./env/lib/python3.8/site-packages (from aider-chat) (1.9.2)
Requirement already satisfied: pytest==7.3.1 in ./env/lib/python3.8/site-packages (from aider-chat) (7.3.1)
Requirement already satisfied: tiktoken==0.4.0 in ./env/lib/python3.8/site-packages (from aider-chat) (0.4.0)
Requirement already satisfied: configargparse in ./env/lib/python3.8/site-packages (from aider-chat) (1.7)
Requirement already satisfied: PyYAML in ./env/lib/python3.8/site-packages (from aider-chat) (6.0.1)
Requirement already satisfied: backoff==2.2.1 in ./env/lib/python3.8/site-packages (from aider-chat) (2.2.1)
Requirement already satisfied: networkx==3.1 in ./env/lib/python3.8/site-packages (from aider-chat) (3.1)
Requirement already satisfied: diskcache==5.6.1 in ./env/lib/python3.8/site-packages (from aider-chat) (5.6.1)
Requirement already satisfied: numpy==1.24.3 in ./env/lib/python3.8/site-packages (from aider-chat) (1.24.3)
Requirement already satisfied: scipy==1.10.1 in ./env/lib/python3.8/site-packages (from aider-chat) (1.10.1)
Requirement already satisfied: jsonschema==4.17.3 in ./env/lib/python3.8/site-packages (from aider-chat) (4.17.3)
Requirement already satisfied: sounddevice==0.4.6 in ./env/lib/python3.8/site-packages (from aider-chat) (0.4.6)
Requirement already satisfied: soundfile==0.12.1 in ./env/lib/python3.8/site-packages (from aider-chat) (0.12.1)
Requirement already satisfied: importlib-resources>=1.4.0 in ./env/lib/python3.8/site-packages (from jsonschema==4.17.3->aider-chat) (6.4.0)
Requirement already satisfied: pkgutil-resolve-name>=1.3.10 in ./env/lib/python3.8/site-packages (from jsonschema==4.17.3->aider-chat) (1.3.10)
Requirement already satisfied: pyrsistent!=0.17.0,!=0.17.1,!=0.17.2,>=0.14.0 in ./env/lib/python3.8/site-packages (from jsonschema==4.17.3->aider-chat) (0.20.0)
Requirement already satisfied: iniconfig in ./env/lib/python3.8/site-packages (from pytest==7.3.1->aider-chat) (2.0.0)
Requirement already satisfied: packaging in ./env/lib/python3.8/site-packages (from pytest==7.3.1->aider-chat) (24.1)
Requirement already satisfied: pluggy<2.0,>=0.12 in ./env/lib/python3.8/site-packages (from pytest==7.3.1->aider-chat) (1.5.0)
Requirement already satisfied: exceptiongroup>=1.0.0rc8 in ./env/lib/python3.8/site-packages (from pytest==7.3.1->aider-chat) (1.2.1)
Requirement already satisfied: tomli>=1.0.0 in ./env/lib/python3.8/site-packages (from pytest==7.3.1->aider-chat) (2.0.1)
Requirement already satisfied: typing-extensions<5.0,>=4.0.0 in ./env/lib/python3.8/site-packages (from rich==13.3.5->aider-chat) (4.12.2)
Requirement already satisfied: CFFI>=1.0 in ./env/lib/python3.8/site-packages (from sounddevice==0.4.6->aider-chat) (1.16.0)
Requirement already satisfied: regex>=2022.1.18 in ./env/lib/python3.8/site-packages (from tiktoken==0.4.0->aider-chat) (2024.5.15)
Requirement already satisfied: pycparser in ./env/lib/python3.8/site-packages (from CFFI>=1.0->sounddevice==0.4.6->aider-chat) (2.22)
Requirement already satisfied: zipp>=3.1.0 in ./env/lib/python3.8/site-packages (from importlib-resources>=1.4.0->jsonschema==4.17.3->aider-chat) (3.19.2)`

**Comment by paul-gauthier** [comment]
My guess is there's something wrong in your local shell or local python environment that's preventing you from installing/running the new version. This will be hard for me to debug.

You can try this:

```
python -m pip install --upgrade aider-chat
python -m aider
```

That should announce that it is running aider v0.40.0.

**Comment by paul-gauthier** [comment]
I'm going to close this issue for now, but feel free to add a comment here and I will re-open or file a new issue any time.
